src/shogun/statistical_testing/internals/Block.cpp

```diff
@@ -48,7 +48,9 @@ Block::Block(CFeatures* feats, index_t index, index_t size) : m_feats(feats)
 
 	// create a shallow copy and subset current block separately
 	CFeatures* block=FeaturesUtil::create_shallow_copy(feats);
+#ifdef USE_REFERENCE_COUNTING
 	ASSERT(block->ref_count()==0);
```

Comment:
Agreed. So, I think 
- this ASSERT should itself be a unit-test (because if someone changes the impl of `FeaturesUtil::create_shallow_copy` later, e.g. `SG_REF` it before returning it, (s)he should be alerted cause it might break things)
- Actually I should move this thing to be a part of the `CFeatures` API maybe. This functionality has no reason to stay within statistical testing hierarchy.

Let me know what you think.